dlib/simd/simd4f.h

```diff
@@ -524,6 +524,11 @@ namespace dlib
 
 // ----------------------------------------------------------------------------------------
 
+#if defined(DLIB_HAVE_VSX) && !defined(vec_rsqrt)
+    extern inline __vector float __attribute__((always_inline)) vec_rsqrt(const __vector float& a)
+    { return vec_div((__vector float){1, 1, 1, 1}, vec_sqrt(a)); }
```

Comment:
Why isn't this `vector float` instead of `__vector vector`?

Reply:
Same reason as in e.g. https://ffmpeg.org/pipermail/ffmpeg-devel/2019-August/247682.html - a namespace collision.